hs-bindgen/src/HsBindgen/Backend/PP/Translation.hs

```diff
@@ -78,12 +78,12 @@ newtype HsModuleOpts = HsModuleOpts {
     }
   deriving stock (Show)
 
-translate :: HsModuleOpts -> C.Header -> HsModule
-translate HsModuleOpts{..} header =
+translate :: TranslationOpts -> HsModuleOpts -> C.Header -> HsModule
```

Comment:
I think this function is out of place. Something like that could exist in `Lib` module, but here it unnecessary imports stuff from previous steps. I'd rather not have it in this module. Let `Transalation` modules just do the translations. Larger pipelines to be assembled elsewhere.

Reply:
I'm not sure I follow. Are you saying that you'd prefer to have a separate `HsModule -> HsModule` step that adds things like instances that we want to be derived?

Reply:
So, the code in `Backend` modulespace is a mess.

Everywhere else `Translation` modules contain code to translate from previous representation to the next one.

In PP code, the `Translation` module has that, but also compound functions.

In TH code, the translation is in `TH` module and compound function in `TH.Translate`.

Again, it's a mess.

I think that compound functions should only be in `Lib` which naturally imports everything. And have `Translation` modules *just do the one step translations*.

You may cleanup this in this PR, or I'll cleanup it afterwards. The current state is not good.

Reply:
Ok, I'm not entirely sure what you envision here, so feel free to submit a PR with a proposed new structure.